File: app/src/main/java/com/ragdroid/dahaka/app/SchedulerModule.java
package com.ragdroid.dahaka.app;

import com.ragdroid.dahaka.util.BaseSchedulerProvider;
import com.ragdroid.dahaka.util.SchedulerProvider;

import dagger.Module;
import dagger.Provides;

/**
 * Created by garimajain on 24/08/17.
 */
@Module
public class SchedulerModule {

    @Provides
    BaseSchedulerProvider providerSchedulerProvider(SchedulerProvider provider) {
        return provider;
    }

}

File: app/src/main/java/com/ragdroid/dahaka/app/SchedulerModule.kt
package com.ragdroid.dahaka.app

import com.ragdroid.dahaka.util.BaseSchedulerProvider
import com.ragdroid.dahaka.util.SchedulerProvider

import dagger.Module
import dagger.Provides

/**
 * Created by garimajain on 24/08/17.
 */
@Module
class SchedulerModule {

    @Provides
    internal fun providerSchedulerProvider(provider: SchedulerProvider): BaseSchedulerProvider {
        return provider
    }

}
